addons/block_code/ui/picker/categories/category_factory.gd

```diff
@@ -204,7 +204,7 @@ static func get_general_blocks() -> Array[Block]:
 
 	b = BLOCKS["control_block"].instantiate()
 	b.block_formats = ["repeat {number: INT}"]
-	b.statements = ["for i in {number}:"]
+	b.statements = ["for i__ in {number}:"]
```

Comment:
Yup, needed for nested loops. If I pass the instruction tree into the `get_instruction_node()` method, it might be easier to handle the going out of scope in which case you could use the same variable names. I just tried to get it working in the least amount of changes, since we'll likely have to rewrite it for the new blocks anyways.

Reply:
Yeah, good enough for now. I just wanted to make sure I was understanding correctly.